test/common_utils.py

```diff
@@ -388,6 +388,11 @@ def assertEqual(self, x, y, prec=None, message='', allow_inf=False):
             self.assertEqual(x.item(), y, prec, message, allow_inf)
         elif isinstance(y, torch.Tensor) and isinstance(x, Number):
             self.assertEqual(x, y.item(), prec, message, allow_inf)
+        elif isinstance(x, torch.BoolTensor) and isinstance(y, torch.BoolTensor):
+            self.assertEqual(x.storage(), y.storage())
```

Comment:
I don't think this is right. If you have a BoolTensor that is a view, storage() will lose that information and give you an incorrect comparison. Better to adjust the code below to take some different strategy with bool is a dtype (and thus we cannot do a subtraction)